spacy/tokens/doc.pyx

```diff
@@ -158,17 +158,38 @@ cdef class Doc:
             raise ValueError(Errors.E046.format(name=name))
         return Underscore.doc_extensions.pop(name)
 
-    def __init__(self, Vocab vocab, words=None, spaces=None, user_data=None):
+    def __init__(
+        self,
+        Vocab vocab,
+        words=None,
+        spaces=None,
+        user_data=None,
```

Comment:
I wonder if we could make the `user_data` keyword-only 🤔 I doubt it's used much in user code and we haven't really been using it either. So I think it'd be safe?

Reply:
I didn't want to change it, but it wasn't even in the docs.

Reply:
If it was undocumented, then let's just go or it and make it keyword-only!